lib/helpers.py

```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+"""## Import needed packages
+
+This notebook/script needs pandas and scipy for analysis and boto to access data store on S3.
+"""
+
+from datetime import datetime
+import pandas as pd
+import numpy as np
+import xxhash
+import re
+import os
+import gzip
+import scipy
+import scipy.stats
+import s3fs
+from google.colab import files
+from IPython.display import display  # so we can run this as script as well
+
+"""## Data loading helpers
+Define a few helper functions to load and cache data.
+"""
+
+VERSION = '5.4.13'
+cache_folder = "cache-v{0}".format(VERSION)
+
+# constants for groups
+TEST = True
+CONTROL = False
+
+# use more memory efficient types for ts,user_id and ab_test_group
+def improve_types(df):
+    df['ts'] = pd.to_datetime(df['ts'])
+    df['ts'] = (df['ts'].astype('int64') / 1e9).astype('int32')
+    df['user_id'] = df['user_id'].apply(xxhash.xxh64_intdigest).astype('int64')
+    df['ab_test_group'] = df['ab_test_group'].transform(lambda g: g == 'test')
+    return df
+
+def path(customer, audience):
+    return "s3://remerge-customers/{0}/uplift_data/{1}".format(customer, audience)
+
+# only keep rows where the event is a revenue event and drop the partner_event
+# column afterwards 
+def extract_revenue_events(df, revenue_event):
+    df = df[df.partner_event == revenue_event]
+    return df.drop(columns=['partner_event'])
+    
+# parquet save and load helper
+def to_parquet(df, filename):
+    df.to_parquet(filename, engine='pyarrow')
+
+# a little "hack" to convert old file on the fly
```

Comment:
Maybe more descriptive docstrings is good idea? A description of the behavior of the function and its parameters would be very useful.